test/python/transpiler/test_trivial_layout.py

```diff
@@ -47,6 +49,24 @@ def test_3q_circuit_5q_coupling(self):
         for i in range(3):
             self.assertEqual(layout[qr[i]], i)
 
+    def test_3q_circuit_5q_coupling_with_target(self):
+        """Test finds trivial layout for 3q circuit on 5q device."""
+        qr = QuantumRegister(3, "q")
+        circuit = QuantumCircuit(qr)
+        circuit.cx(qr[1], qr[0])
+        circuit.cx(qr[0], qr[2])
+        circuit.cx(qr[1], qr[2])
+
+        dag = circuit_to_dag(circuit)
+        target = Target()
+        target.add_instruction(CXGate(), {tuple(edge): None for edge in self.cmap5})
+        pass_ = TrivialLayout(target=target)
+        pass_.run(dag)
+        layout = pass_.property_set["layout"]
+
+        for i in range(3):
+            self.assertEqual(layout[qr[i]], i)
```

Comment:
Pre-existing, but having worked on Sabre enough, I have hatred in my heart for this kind of list testing.  Better to do
```python
self.assertEqual([layout[qr[i]] for i in range(3)], list(range(3)))
```
so if there's a failure we can see _all_ the failures at once.

No need to change that in this PR, though - it's a larger unrelated gripe for me.